### PR TITLE
DOC: Add release note for Highway VQSort on AArch64

### PR DESCRIPTION
**Pull request by Mousius**

### CODE REVIEW

**Comment by Mousius** [comment]
@seiko2plus does this fit in the release notes? My first attempt 😸

File: doc/release/upcoming_changes/24018.improvement.rst [hunk 1]
```diff
@@ -0,0 +1,7 @@
+AArch64 quicksort speed measured to be improved by up to 16x using Highway's VQSort
+-----------------------------------------------------------------------------------
```
Comment (charris) [inline doc/release/upcoming_changes/24018.improvement.rst:2]: I'd shorten that a bit:
```
AArch64 quicksort speed improved by using Highway's VQSort
```

File: doc/release/upcoming_changes/24018.improvement.rst [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+The first introduction of the Google Highway library, using the highly optimized VQSort:
+https://opensource.googleblog.com/2022/06/Vectorized%20and%20performance%20portable%20Quicksort.html
+
+Initially, this targets AArch64, but Highway supports many platforms and algorithms, which can be enabled in the future.
```
Comment (charris) [inline doc/release/upcoming_changes/24018.improvement.rst:7]: I'd shorten this a bit also:
```
The first introduction of the Google Highway library, using VQSort on AArch64. 
Execution time is improved by up to 16x in some cases, see the PR for benchmark
results. Extensions to other platforms will be done in the future.
```
The release notes come with a link to the PR, so the benchmarks can be seen there.

**Comment by Mousius** [comment]
Thanks for your help with the words @charris 😸

**Comment by charris** [comment]
Thanks @Mousius .

**Pull request merged**